spec/lib/survey/gateway/notifications_spec.rb

```diff
@@ -0,0 +1,57 @@
+describe Survey::Gateway::Notifications do
+  let(:user) { FactoryBot.create(:user_details) }
+  let(:mobile_user) { FactoryBot.create(:user_details, :sms) }
+
+  let(:notify_email_url) { "https://api.notifications.service.gov.uk/v2/notifications/email" }
+  let(:notify_mobile_url) { "https://api.notifications.service.gov.uk/v2/notifications/sms" }
+
+  let(:notify_email_request) do
+    {
+      email_address: user.contact,
+      template_id: "email-mock-template",
+    }
+  end
+
+  let(:notify_mobile_request) do
+    {
+      phone_number: mobile_user.contact,
+      template_id: "mobile-mock-template",
+    }
+  end
+
+  let(:notify_mobile_stub) do
+    stub_request(:post, notify_mobile_url)
+      .with(body: notify_mobile_request)
+      .to_return(status: 200, body: {}.to_json)
+  end
```

Comment:
This block (notify_mobile_stub) should probably be moved to the "when the user signed up by mobile" context.